docs/development/release.rst

```diff
@@ -6,30 +6,47 @@
 How to make a Gammapy release
 *****************************
 
-This page contains step-by-step instructions how to make a Gammapy release. We mostly follow the
-`Astropy release instructions <https://docs.astropy.org/en/latest/development/maintainers/releasing.html#release-procedure-for-the-astropy-core-package>`__
-and just lists the additional required steps.
+This page contains step-by-step instructions for how to make a Gammapy release. The general procedure can
```

Comment:
I would keep the mention of the astropy release scheme saying that our procedure is inspired by that of astropy and follows similar steps and refer to their doc here.